pcsx2 crashes after starting with 'Illegal instruction' if built with gcc-7.1.0

***PCSX2 version:***
`pcsx2-2017.05.08_9b0b32533_master-x86_64-1_git`
`wxGTK3-3.0.3-x86_64-1_SBo`

***PCSX2 options:***
Occurs even after the first start up without a configuration file.

***Description of the issue:***
If wxGTK3 is built without `-with-sdl` pcsx2 will crash after starting with `Illegal instruction`. If there is no configuration file it will crash after hitting 'Next' in the configuration wizard. Further more pcsx2 will not build if wxGTK3 is built with SDL2 and will also fail to build with `DSDL2_API=FALSE`.
```
$ PCSX2 
Interface is initializing.  Entering Pcsx2App::OnInit!
Applying operating system default language...
Command line parsing...
Command line parsed!
Creating UserLocalData folder: /home/orbea/.config/PCSX2
Illegal instruction
```
GDB log: https://pastebin.com/BTiaHsWp

***How to reproduce the issue:***
1. Build wxGTK3 without SDL(2).
1. Build pcsx2 and start it.

***Last known version to work:***
```
ef3435441f7a86d2062f2eeaa0508455c296feaf is the first bad commit
commit ef3435441f7a86d2062f2eeaa0508455c296feaf
Author: Gregory Hainaut <gregory.hainaut@gmail.com>
Date:   Sat Apr 15 14:57:00 2017 +0200

    onepad: create a legacy version for SDL1
    
    The legacy version is the version 1.3
    
    The new version is the version 2.0. It would be based on SDL2 only
    
    Distribution information.
    If you link wxWidget with SDL, you will need to be sure it is SDL2 or upgrade.
    
    You can build wxWidget with SDL2 with the following trick
    
    export SDL_CONFIG=/usr/bin/sdl2-config
    ./configure --with-sdl .....

:040000 040000 a289d4b0a8efaf6380daaf5b7023de6689dc9d11 cb4e544be1dbed9ca1ce49f4e5a7e6e9b9747a22 M	cmake
:040000 040000 73ec2b83821437382b3614f5a6d1cbf643c4b49a eaca65a43c10adaf0fda260a1ba6850b75dd8480 M	plugins
```
ef3435441f7a86d2062f2eeaa0508455c296feaf

So I attempted to follow the advice in the commit message and built wxGTK3 with SDL2 and got this. During the wxGTK3 compile I did see it include SDL2, but maybe wxGTK3 version is too old?
```
-- Found GTK2_GTK: /usr/lib/libgtk-x11-2.0.so  
CMake Error at cmake/ApiValidation.cmake:23 (message):
  WxWidget is linked to SDL (wxUSE_LIBSDL) and it is likely SDL1.2.

          Unfortunately you try to build PCSX2 with SDL2 support which is not compatible
          Please use -DSDL2_API=FALSE
Call Stack (most recent call first):
  cmake/SearchForStuff.cmake:194 (WX_vs_SDL)
  CMakeLists.txt:61 (include)


-- Configuring incomplete, errors occurred!
See also "/tmp/SBo/pcsx2/build/CMakeFiles/CMakeOutput.log".
```
So I then tried `-DSDL2_API=FALSE` and got this instead...
```
-- Found GTK2_GTK: /usr/lib/libgtk-x11-2.0.so  
CMake Error at cmake/SelectPcsx2Plugins.cmake:20 (message):
  Skip build of onepad: miss some dependencies:check these libraries -> sdl2,
  X11, gtk2
Call Stack (most recent call first):
  cmake/SelectPcsx2Plugins.cmake:228 (print_dep)
  CMakeLists.txt:62 (include)

-- Configuring incomplete, errors occurred!
See also "/tmp/SBo/pcsx2/build/CMakeFiles/CMakeOutput.log".
```
Is there a way out of this? 

***PC specifications:***
OS: `Slackware64-current`
CPU: `amd fx 6350`
GPU: `GTX 780 Ti`

```
-- Found GTK2_GTK: /usr/lib/libgtk-x11-2.0.so  
CMake Error at cmake/ApiValidation.cmake:23 (message):
  WxWidget is linked to SDL (wxUSE_LIBSDL) and it is likely SDL1.2.

          Unfortunately you try to build PCSX2 with SDL2 support which is not compatible
          Please use -DSDL2_API=FALSE
```
Hum, it means that wx still use SDL. At least one wx version installed on your system. And cmake pick-up the wrong one.


Yes it is expected that you can't build onepad2 without SDL2. However, you can still use onepad-legacy

--- a/plugins/GSdx/GS.cpp
+++ b/plugins/GSdx/GS.cpp
@@ -128,6 +128,7 @@ EXPORT_C_(int) GSinit()
 	// const type qualifier from all the affected variables.
 	theApp.Init();
 
+	GSUtil::Init();
 	GSBlock::InitVectors();
 	GSClut::InitVectors();
 #ifdef ENABLE_OPENCL

--- a/plugins/GSdx/GSUtil.cpp
+++ b/plugins/GSdx/GSUtil.cpp
@@ -102,7 +102,8 @@ public:
 	uint32 CompatibleBitsField[64][2];
 	uint32 SharedBitsField[64][2];
 
-	GSUtilMaps()
+	// Defer init to avoid AVX2 illegal instructions
+	void Init()
 	{
 		PrimClassField[GS_POINTLIST] = GS_POINT_CLASS;
 		PrimClassField[GS_LINELIST] = GS_LINE_CLASS;
@@ -162,6 +163,11 @@ public:
 	}
 
 } s_maps;
+
+void GSUtil::Init()
+{
+	s_maps.Init();
+}
 
 GS_PRIM_CLASS GSUtil::GetPrimClass(uint32 prim)
 {

--- a/plugins/GSdx/GSUtil.h
+++ b/plugins/GSdx/GSUtil.h
@@ -37,6 +37,8 @@ struct OCLDeviceDesc
 class GSUtil
 {
 public:
+	static void Init();
+
 	static const char* GetLibName();
 
 	static GS_PRIM_CLASS GetPrimClass(uint32 prim);

